### ISSUE TITLE
Questions about vae.py

### ISSUE THREAD

**Issue by ifangcheng** [thu-ml/zhusuan#93]
regarding to vae.py, I have two questions：

1.   what's the considerations to use  zs.Implicit and  zs.Empirical  ?  any benefits from using these two API?
x_mean = zs.Implicit("x_mean", tf.sigmoid(x_logits), group_ndims=1)      in line 27
x = zs.Empirical('x', tf.int32, (None, x_dim))   in line 34

2.   why the   n_particles set for evaluating  test_lb  and test_ll differs so much (1 vs 1000)?
       test_lb = sess.run(lower_bound, feed_dict={x: test_x_batch,  n_particles: 1})   in line 125
       test_ll = sess.run(is_log_likelihood,feed_dict={x: test_x_batch, n_particles: 1000})  in line 128

**Comment by thjashin** [comment]
> what's the considerations to use zs.Implicit and zs.Empirical ? any benefits from using these two API?

This is an optional feature. Some people would like to be more abstract about the model construction. And they don't like passing to or returning from the model function with additional parameters. This feature helps so that you can query the model about tensor you want.

**Comment by thjashin** [comment]
> why the n_particles set for evaluating test_lb and test_ll differs so much (1 vs 1000)?

For test_lb, this is kind of arbitrary, in principle it should be large to be more accurate. For test_ll, it has to be large. Because the log likelihood is estimated using importance sampling, which is inaccurate when the number of samples is not sufficiently large.

**Comment by ifangcheng** [comment]
@thjashin  sounds reasonable！ many thanks！
